src/appleseed/renderer/kernel/rendering/final/adaptivetilerenderer.cpp

```diff
@@ -298,24 +292,23 @@ namespace
             create_rendering_blocks(rendering_blocks, padded_tile_bbox, framebuffer->get_crop_window());
 
             // First uniform pass based on adaptiveness parameter.
-            if (m_params.m_adaptiveness < 1.0f)
+            if (m_params.m_min_samples > 0)
             {
+                const size_t batch_size = m_params.m_max_samples > 0 ?
+                    min(m_params.m_min_samples, m_params.m_max_samples) : m_params.m_min_samples;
+
                 for (size_t i = 0; i < rendering_blocks.size(); ++i)
                 {
                     PixelBlock& pb = rendering_blocks.front();
                     rendering_blocks.pop_front();
```

Comment:
I'm not sure this is correct. Is `pb` guaranteed to still be valid once you called `pop_front()`? Intuitively I would say no.

Reply:
What do you mean by valid?

Reply:
It seems that it's potentially a reference to a destroyed object.

Reply:
Ah I see, I should copy it ! I still make this mistake a lot. Thanks

Reply:
Yeah this is clearly a bug, or a potential serious bug:

https://stackoverflow.com/questions/6244931/reference-front-and-pop-front

Reply:
Demo of the problem with VC++:
http://rextester.com/FPUA29279

Same problem with clang. However, interestingly it works fine with gcc. Obviously we're deep into undefined territory here so compilers are free to do whatever they want.

Reply:
Wow thanks, it's weird i didn't noticed any issue so far